### ISSUE TITLE
UnitsWarnings in downloading products with Astropy v5.1

### ISSUE THREAD

**Issue by keyuxing** [lightkurve/lightkurve#1216]
#### Problem description

Warning messages are shown when reading light curves.

#### Example
```
>>> import lightkurve as lk
>>> lk.search_lightcurve('tic77031414', sector=2, exptime=120, author='spoc').download()

WARNING: UnitsWarning: 'BJD - 2457000, days' did not parse as fits unit: At col 0, Unit 'BJD' 
not supported by the FITS standard.  If this is meant to be a custom unit, define it with 'u.def_unit'. 
To have it recognized inside a file reader or other code, enable it with 'u.add_enabled_units'. 
For details, see https://docs.astropy.org/en/latest/units/combining_and_defining.html [astropy.units.core]

WARNING: UnitsWarning: 'e-/s' did not parse as fits unit: At col 0, Unit 'e' not supported by the 
FITS standard.  If this is meant to be a custom unit, define it with 'u.def_unit'. To have it recognized 
inside a file reader or other code, enable it with 'u.add_enabled_units'. For details, see 
https://docs.astropy.org/en/latest/units/combining_and_defining.html [astropy.units.core]

WARNING: UnitsWarning: 'pixels' did not parse as fits unit: At col 0, Unit 'pixels' not supported by the 
FITS standard. Did you mean pixel? If this is meant to be a custom unit, define it with 'u.def_unit'. To 
have it recognized inside a file reader or other code, enable it with 'u.add_enabled_units'. For details, 
see https://docs.astropy.org/en/latest/units/combining_and_defining.html [astropy.units.core]
```

#### Environment

-  lightkurve v2.2.1
-  astropy v5.1

**Comment by orionlee** [comment]
The new `UnitsWarning` in Astropy 5.1 is a nuisance, but does not affect actual functionalities.

Astropy 5.1 seems to be more aggressive in issuing warning. The warning stems from reading the actual TESS FITS file content in: 

https://github.com/lightkurve/lightkurve/blob/71c4037936578ba0ccd41f387a1def38f1fadb3e/src/lightkurve/io/generic.py#L40

The existing codes convert the non-standard units in FITS file to standard ones after the `read()` call, e.g., 
https://github.com/lightkurve/lightkurve/blob/71c4037936578ba0ccd41f387a1def38f1fadb3e/src/lightkurve/io/generic.py#L52-L53

One way to deal with it is to catch the warning in the `Table.read()` call. 

We could possibly use `u.def_unit` and `u.add_enabled_units` suggested in the warning message to make Astropy recognize them. However, the impact would be global so we probably should stick with catching the warning.

**Comment by keyuxing** [comment]
Thanks for your explanation!

I just created a PR (#1226) trying to resolve this issue. Hope this helps.

**Comment by orionlee** [comment]
I'm reopening the issue to track a related issue.

I suspect the `UnitsWarnings` could also show up when using `CBVCorrector`. It also uses `Table.read()` calls to open CBV fits files:

https://github.com/lightkurve/lightkurve/blob/f898fad53487deef74ade98a348441232ac3d803/src/lightkurve/correctors/cbvcorrector.py#L1444

https://github.com/lightkurve/lightkurve/blob/f898fad53487deef74ade98a348441232ac3d803/src/lightkurve/correctors/cbvcorrector.py#L1622


I have not yet verified if that is indeed the case.

**Comment by keyuxing** [comment]
Yes. The `UnitsWarning` does show up when using `CBVCorrector`:

```
>>> from lightkurve import search_lightcurve
>>> from lightkurve.correctors import CBVCorrector
>>> lc = search_lightcurve('TIC 99180739', author='SPOC', sector=10).download()
>>> cbvCorrector = CBVCorrector(lc)
Downloading https://archive.stsci.edu/missions/tess/ffi/s0010/2019/085/1-1/tess2019085135100-s0010-1-1-0140-s_cbv.fits
|==========================================| 4.7M/4.7M (100.00%)      1m39s
WARNING: UnitsWarning: 'JD - 2457000, days' did not parse as fits unit: At col 0, Unit 'JD' not 
supported by the FITS standard.  If this is meant to be a custom unit, define it with 'u.def_unit'. 
To have it recognized inside a file reader or other code, enable it with 'u.add_enabled_units'. 
For details, see https://docs.astropy.org/en/latest/units/combining_and_defining.html [astropy.units.core]
```

**Comment by barentsen** [comment]
The warnings triggered via `CBVCorrector` should also be fixed now via #1229.
